### PR TITLE
Feature: Implemented Multiple Vectors

### PR DESCRIPTION
**Pull request by harshaxnim**
With these code changes, users can now add multiple vectors and multiple expressions.

The buttons to add the vectors/expressions are appended at the bottom of the current UI. The expressions right now are very simple; users can add, subtract, or cross multiply only two vectors.

I might add complex expression evaluation at a later point of time if users seem to be wanting it.

There are a lot of changes, so I'll try to comment on each change for the ease of your review.

Here's a screenshot of how it looks after the changes:

Screenshot 2019-08-04 at 3 28 28 PM

### CODE REVIEW

File: demos/3d-vector-plotter/demo.js [hunk 1]
```diff
@@ -100,46 +104,172 @@
...
+			vCross[2] = v1[0] * v2[1] - v1[1] * v2[0];
+			return vCross;
+		},
+
 		update: function(e){
 
-			this.v1 = this.ui.v1.value.substring(1, this.ui.v1.value.length-1).split(","); 
-			this.v2 = this.ui.v2.value.substring(1, this.ui.v2.value.length-1).split(",");
-			this.v3 = this.ui.v3.value.substring(1, this.ui.v3.value.length-1).split(",");
+			if (e == "addVector") {
```
Comment (harshaxnim) [inline demos/3d-vector-plotter/demo.js:193]: handling the add vector button.

File: demos/3d-vector-plotter/demo.js [hunk 2]
```diff
@@ -100,46 +104,172 @@
...
+										type: "userInputString"
+									};
+				this.addUIElement(vecName);
+
+				// push it to the intermediate variable and push name to list for record
+				this[vecName] = [1,2,3];
+				this.vecList.push(vecName);
+			}
+
+			else if (e == "addExpression") {
```
Comment (harshaxnim) [inline demos/3d-vector-plotter/demo.js:217]: handling the add expression button.

File: demos/3d-vector-plotter/demo.js [hunk 3]
```diff
@@ -100,46 +104,172 @@
...
+									};
+				this.addUIElement(vecName);
+				
+				// push it to the intermediate variable and push name to list for record
+				this[vecName] = this.evaluateExp('v1 x v2');
+				this.vecList.push(vecName);
+			}
+
+
+			// update vector values
```
Comment (harshaxnim) [inline demos/3d-vector-plotter/demo.js:242]: reading the values from the UI

File: js/humanColors.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
+Colors = {
```
Comment (harshaxnim) [inline js/humanColors.js:1]: For reasonable colours for humans with good contrast between them, I realised this is better than picking a random colour.

**Comment by harshaxnim** [comment]
Resolved conflicts! 😄

**Comment by edwardball** [comment]
Brilliant, thank you! This is a great feature, one that many people have asked about so it's excellent to have this functionality now. All your pull requests are now merged and deployed on the site :)

**Comment by harshaxnim** [comment]
haha, thank you! glad to have helped! :)

**Pull request merged**